core/src/main/java/org/apache/carbondata/core/carbon/querystatistics/QueryStatisticsRecorder.java

```diff
@@ -61,14 +67,275 @@ public QueryStatisticsRecorder(String queryId) {
    */
   public synchronized void recordStatistics(QueryStatistic statistic) {
     queryStatistics.add(statistic);
+    // refresh query Statistics Map
+    String key = statistic.getQueryId();
+    if (!StringUtils.isEmpty(key)) {
+      // 240954528274124_0 and 240954528274124 is the same query id
+      key = key.substring(0, 15);
+    }
+    if (queryStatisticsMap.get(key) != null) {
```

Comment:
i think 240954528274124_0  means queryid for each segment? yes?
each executor will print whole time for all segment 
if we have 3 executors, 2 segments, print like:
executor1 log:
|        +--------------------+                     +----------------------+
|        |     Dictionary load|                     |     0.002  0.001 |
|        +--------------------+                     +----------------------+

executor 2 log:
|        +--------------------+                     +----------------------+
|        |     Dictionary load|                     |     0.003  0.002 |
|        +--------------------+                     +----------------------+

executor3 log:
|        +--------------------+                     +----------------------+
|        |     Dictionary load|                     |     0.001  0.003 |
|        +--------------------+                     +----------------------+